File: go/0122_best_time_to_buy_and_sell_stock_ii.go
func maxProfit(prices []int) int {
	profit, sell := 0, false

	for i := 0; i < len(prices); i++ {
		if sell {
			profit, sell = profit+prices[i]-prices[i-1], false
		}
		sell = i < len(prices)-1 && prices[i] < prices[i+1]
	}
	return profit
}

File: python/0122_best_time_to_buy_and_sell_stock_II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        res = 0
        i = 0
        localmin, localmax = 0, 0
        while i < len(prices):
            while i < len(prices):
                localmin = i
                if i < len(prices) - 1 and prices[i + 1] > prices[i]:
                    i += 1
                    break
                i += 1

            while i < len(prices):
                localmax = i
                if i < len(prices) - 1 and prices[i + 1] < prices[i]:
                    i += 1
                    break
                i += 1

            if localmin < localmax:
                res += prices[localmax] - prices[localmin]
        return res